zk_toolbox/crates/zk_supervisor/src/commands/test/integration.rs

```diff
@@ -1,31 +1,76 @@
+use std::{collections::HashMap, path::PathBuf, vec};
+
+use anyhow::Context;
 use common::{cmd::Cmd, config::global_config, logger, spinner::Spinner};
-use config::EcosystemConfig;
+use config::{ChainConfig, EcosystemConfig};
+use ethers::{
+    signers::{coins_bip39::English, MnemonicBuilder, Signer},
+    types::H160,
+    utils::hex,
+};
+use serde::Deserialize;
 use xshell::{cmd, Shell};
 
 use super::args::integration::IntegrationArgs;
 use crate::messages::{
-    msg_integration_tests_run, MSG_INTEGRATION_TESTS_BUILDING_CONTRACTS,
+    msg_integration_tests_run, MSG_CHAIN_NOT_FOUND_ERR, MSG_INTEGRATION_TESTS_BUILDING_CONTRACTS,
     MSG_INTEGRATION_TESTS_BUILDING_DEPENDENCIES, MSG_INTEGRATION_TESTS_RUN_SUCCESS,
 };
 
 const TS_INTEGRATION_PATH: &str = "core/tests/ts-integration";
+const TEST_WALLETS_PATH: &str = "etc/test_config/constant/eth.json";
 const CONTRACTS_TEST_DATA_PATH: &str = "etc/contracts-test-data";
+const AMOUNT_FOR_DISTRIBUTION_TO_WALLETS: u128 = 1000000000000000000000;
 
-pub fn run(shell: &Shell, args: IntegrationArgs) -> anyhow::Result<()> {
+pub async fn run(shell: &Shell, args: IntegrationArgs) -> anyhow::Result<()> {
     let ecosystem_config = EcosystemConfig::from_file(shell)?;
+    let chain_config = ecosystem_config
+        .load_chain(global_config().chain_name.clone())
+        .context(MSG_CHAIN_NOT_FOUND_ERR)?;
     shell.change_dir(ecosystem_config.link_to_code.join(TS_INTEGRATION_PATH));
 
     logger::info(msg_integration_tests_run(args.external_node));
 
-    build_repository(shell, &ecosystem_config)?;
-    build_test_contracts(shell, &ecosystem_config)?;
+    if !args.no_deps {
+        build_repository(shell, &ecosystem_config)?;
+        build_test_contracts(shell, &ecosystem_config)?;
+    }
+
+    let wallets_path: PathBuf = ecosystem_config.link_to_code.join(TEST_WALLETS_PATH);
+    let test_wallets: TestWallets = serde_json::from_str(shell.read_file(&wallets_path)?.as_ref())
+        .context("Impossible to deserialize test wallets")?;
+    let test_wallet_id: String = format!("test_mnemonic{}", chain_config.id + 1);
+
+    let phrase = test_wallets
+        .wallets
+        .get(test_wallet_id.as_str())
+        .unwrap()
+        .as_str();
+
+    let mnemonic = MnemonicBuilder::<English>::default()
+        .phrase(phrase)
+        .derivation_path(test_wallets.base_path.as_str())
```

Comment:
Are you sure it's better to change mnemonic, rather than changing basic_path, in your case we are limited by 10 addresses